docs/concepts/stateful-stream-processing.md

```diff
@@ -242,6 +246,42 @@ updates to that state.
 See [Restart Strategies]({% link dev/task_failure_recovery.md
 %}#restart-strategies) for more information.
 
+### Unaligned Checkpointing
+
+Starting with Flink 1.11, checkpointing can also be performed unaligned.
+The basic idea is that checkpoints can overtake all in-flight data as long as 
+the in-flight data becomes part of the operator state.
+
+<div style="text-align: center">
+  <img src="{% link fig/stream_unaligning.svg %}" alt="Unaligned checkpointing" style="width:100%; padding-top:10px; padding-bottom:10px;" />
+</div>
+
+The figure depicts how an operator handle unaligned checkpoint barriers:
+
+- The operator reacts on the first barrier that is stored in input buffers.
+- It immediately forwards the barrier to the downstream operator by adding it 
+  to the end of the output buffers.
+- The operator marks all overtaken buffers, which are asynchronously stored in
+  the state backend together with the other operator state.
+- It only briefly stops the processing of input to mark the buffers, forward 
+  the barrier and create the snapshot of the other state.
```

Comment:
It reads like a separate step (it's not, right?)
```suggestion
Compared to aligned checkpoints, the operator doesn't need to suspend any of its inputs.
```

Reply:
Yes, I have removed the bullet and made it a summarizing sentence.